### PR TITLE
Add dialog with details on goals component - fixes #249

### LINKED ISSUE #249: Goals on path details and styling
Example: http://staging.unleash.x-team.com/profiles/114634487399949912483

You can see a list of goals. When I click on a goal it would be good for the description of the goal to be visible (maybe appear like in an accordion) and also goals that have the property `achieved` set to true should be distinguishable from others (maybe different color or something?)

### PR DESCRIPTION
**Pull request by jenriquejr**
- [x] Add description dialog for goal card
- [x] Add higher order component for toggle states logic
- [x] Add JSDocs
- [x] Include tests for update components
- [x] Create tests for HOC component

### CODE REVIEW

File: app/components/GoalCard.js [hunk 1]
```diff
@@ -16,14 +27,36 @@ class GoalCard extends Component {
...
+
   render() {
     const { goal } = this.props;
     const achieved = goal.achieved;
     const dueDays = goal.dueDate ? this.daysLeft(goal.dueDate) : 0;
     const goalStyle = Object.assign({}, styles.goal, achieved && styles.achieved);
 
     return (
-      <Paper key={goal.id} style={goalStyle} zDepth={2}>
+      <Paper style={goalStyle} zDepth={2} onClick={() => this.props.toggleOn(DIALOG_TOGGLE)} >
```
Comment (kdemoya) [inline app/components/GoalCard.js]: I had this convention on my mind to always use `onTouchTap` for `MUI` components, not sure if that's justificable, tho...  Any thoughts on `onTouchTap` vs `onClick`?

Comment (jenriquejr) [inline app/components/GoalCard.js]: updated to `onTouchTap` I don't really have a preference here and it seems that `onTouchTap` is good for compatibility with mobile devices

**Comment by jenriquejr** [comment]
ready for review @kdemoya @karolsojko

**Comment by kdemoya** [comment]
Looks good to me 👍

**Comment by karolsojko** [comment]
Good job @jenriquejr - thank you for this contribution 💪  !!

**Pull request merged**